### DISCUSSION TITLE
Add support for Redis storage adapter

### DISCUSSION THREAD

**Discussion by zenire** [spatie/laravel-prometheus#34] [Ideas]
The need for using the Redis storage adapter may exist when users have deployed their Laravel application across multiple servers/instances (for example when using k8s). Using a shared storage adapter will prevent incorrect measurements.

https://github.com/spatie/laravel-prometheus/blob/7aa620d7dae50b27737f43804e4f6c47f7113aa7/src/PrometheusServiceProvider.php#L36 will have to be changed to add support for choosing storage adapters like `InMemory()` and `Redis()`. We can specify the adapter in `config/prometheus.php` and reuse the Laravel Redis connection details or specify a custom connection.

**Comment by Husseinhewehii** [comment]
injecting the Storage adapter is already handler by the package

in config/prometheus.php updating cache Key did the job for me, `'cache' => 'redis'`
